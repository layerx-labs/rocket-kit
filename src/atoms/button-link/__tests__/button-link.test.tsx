import React from 'react';
import { render, screen } from '@testing-library/react';
import { ButtonLink } from '../../..';
import userEvent from '@testing-library/user-event';

describe('Button Link', () => {
  it('renders', () => {
    const { asFragment } = render(
      <ButtonLink url="https://github.com/taikai/taikai-design-system" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('has correct value', () => {
    const buttonText = 'Check the source code';
    render(
      <ButtonLink
        url="https://github.com/taikai/taikai-design-system"
        value={buttonText}
      />
    );
    screen.queryByText(buttonText);
  });

  it('has correct href', () => {
    const buttonText = 'Check the source code';
    const buttonUrl = 'https://github.com/taikai/taikai-design-system';
    render(<ButtonLink url={buttonUrl} value={buttonText} />);
    const anchorElem = screen.getByText(buttonText).closest('a');
    expect(anchorElem).toHaveProperty('href', buttonUrl);
  });

  it('has target _blank and noreferrer', () => {
    const buttonText = 'Check the source code';
    const buttonUrl = 'https://github.com/taikai/taikai-design-system';
    render(<ButtonLink url={buttonUrl} value={buttonText} blank />);
    const anchorElem = screen.getByText(buttonText).closest('a');
    expect(anchorElem).toHaveProperty('target', '_blank');
    expect(anchorElem).toHaveProperty('rel', 'noopener noreferrer');
  });

  it('has empty target if not defined', () => {
    const buttonText = 'Check the source code';
    const buttonUrl = 'https://github.com/taikai/taikai-design-system';
    render(<ButtonLink url={buttonUrl} value={buttonText} />);
    const anchorElem = screen.getByText(buttonText).closest('a');
    expect(anchorElem).toHaveProperty('target', '');
  });

  it('calls action callback', async () => {
    const onClickAction = jest.fn();
    const buttonText = 'Check the source code';
    const buttonUrl = 'https://github.com/taikai/taikai-design-system';
    render(
      <ButtonLink url={buttonUrl} value={buttonText} action={onClickAction} />
    );
    const button = await screen.findByText(buttonText);
    userEvent.click(button);
    expect(onClickAction).toBeCalledTimes(1);
  });
});
