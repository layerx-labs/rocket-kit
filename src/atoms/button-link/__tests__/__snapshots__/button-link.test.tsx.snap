// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Link renders 1`] = `
<DocumentFragment>
  .c0 {
  --button: hsl(186,62%,59%);
  --txt: hsl(0,0%,100%);
  --hover: #2ba8b6;
  --button: hsl(186,62%,59%);
  --hover: #2798a5;
  border-width: 0;
  border-style: solid;
  border-color: var(--button);
  border-radius: 999px;
  background-color: var(--button);
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  min-width: 2.25rem;
  height: 2.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
  text-transform: uppercase;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--hover);
}

.c0 > *:not(:last-child) {
  margin-right: 5px;
}

.c0 span {
  position: relative;
  font-size: 0.75rem;
  font-weight: 700;
  color: var(--txt);
  pointer-events: none;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0 svg {
  width: auto;
  min-width: 1.25rem;
  height: 1.25rem;
  fill: var(--txt);
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

@media (min-width:1280px) {
  .c0 {
    min-width: 2.25rem;
  }
}

<a
    class="c0 button button-link"
    color="primary"
    data-event=""
    data-testid=""
    href="https://github.com/taikai/taikai-design-system"
    rel=""
    target=""
    value=""
  />
</DocumentFragment>
`;
